dascore/io/index/planned.py

```
@@ -671,6 +676,29 @@ def _residual_ranges(residuals) -> dict:
+    states. An output assembled from more than one member, or from a
+    member a trim cuts down, has a size only the loaded patch knows, and
+    states none rather than a member's.
+    """
+    if trims.empty or "_data_size" not in sources.columns:
+        return {}
+    counts = trims.groupby("output_id")["_patch_id"].transform("size").to_numpy()
+    modified = np.asarray(trims.get("_modified", False), dtype=bool)
+    whole = trims[(counts == 1) & ~modified]
+    lookup = sources.drop_duplicates("_patch_id").set_index("_patch_id")["_data_size"]
+    out = {}
+    for output_id, patch_id in zip(whole["output_id"], whole["_patch_id"]):
```

Ruff (0.16.1)

[warning] `zip()` without an explicit `strict=` parameter

Add explicit value for parameter `strict=`

(B905)